### DISCUSSION TITLE
Multi-project runs or nested projects

### DISCUSSION THREAD

**Discussion by tomer-amir-apiiro** [xunit/xunit#2879] [Question]
I want to split a large unit tests project into multiple smaller projects so developers can only build and run the relevant unit tests to what they are doing when they work locally.

The issue is that I still want to run all the unit tests as if they were the big project in our CI for simplicity, coverage and such.

the following command didn't work for me:

```sh
$ dotnet test path/to/subproject1.csproj test path/to/subproject2.csproj
MSBUILD : error MSB1008: Only one project can be specified.
```

same with any glob tricks. I also tried to create a parent project and reference the sub-projects but it ignores the tests in the subprojects.

Is there a way to do this?

**Comment by bradwilson** [comment]
I wrote up some information about the differences between `dotnet test` and `dotnet vstest`, and I think what you want is the latter. It was in the context of parallelization, but the core information will be useful to you: https://github.com/xunit/visualstudio.xunit/issues/191#issuecomment-1763975518
